bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -18,47 +18,624 @@ package v1alpha1
+	// +optional
+	NodeLabels []string `json:"nodeLabels,omitempty"`
+
+	// NodeTaints Registering kubelet with set of taints
+	// +optional
+	NodeTaints []string `json:"nodeTaints,omitempty"`
+
+	// NodeNamePrefix Prefix to the Node Name that CAPI will generate
+	// +optional
+	NodeNamePrefix string `json:"nodeName,omitempty"`
+
+	// ImageCredentialProviderBinDir The path to the directory where credential provider plugin binaries are located
+	//+optional
+	// ImageCredentialProviderBinDir string `json:"imageCredentialProviderBinDir,omitempty"`
+
+	// ImageCredentialProviderConfigMap is a reference to the ConfigMap that contains credential provider plugin config
+	// The configMap should contain a YAML file content + a Path to the Binaries for Credential Provider.
+	//+optional
+	ImageCredentialProviderConfigMap ConfigMapRef `json:"imageCredentialProviderConfigMap,omitempty"`
+
+	// TODO: Remove ContainerRuntimeEndpoint since this feature will probably not be offered by CAPI Bootstrap provider?
+
+	// ContainerRuntimeEndpoint Disable embedded containerd and use alternative CRI implementation
+	// +optional
+	ContainerRuntimeEndpoint string `json:"containerRuntimeEndpoint,omitempty"`
+
+	// Snapshotter Override default containerd snapshotter (default: "overlayfs")
+	// +optional
+	Snapshotter string `json:"snapshotter,omitempty"`
+
+	// TODO: Find a way to handle IP addresses that should be advertised but that RKE2 cannot find on the host (Example: Elastic IPs on Cloud Providers)
+
+	// NodeIp IPv4/IPv6 addresses to advertise for node
+	// +optional
+	//NodeIp string `json:"nodeIp,omitempty"`
+
+	// NodeExternalIp IPv4/IPv6 external IP addresses to advertise for node
+	// +optional
+	// NodeExternalIp string `json:"nodeExternalIp,omitempty"`
+
+	// ResolvConf Kubelet resolv.conf file
+	// +optional
+	ResolvConf string `json:"resolvConf,omitempty"`
+
+	// ProtectKernelDefaults Kernel tuning behavior. If set, error if kernel tunables are different than kubelet defaults.
+	// +optional
+	ProtectKernelDefaults string `json:"protectKernelDefaults,omitempty"`
+
+	// SystemDefaultRegistry Private registry to be used for all system images
+	// +optional
+	SystemDefaultRegistry string `json:"systemDefaultRegistry,omitempty"`
+
+	// Selinux Enable SELinux in containerd
+	// +optional
+	Selinux string `json:"selinux,omitempty"`
+
+	// KubeletPath Override kubelet binary path
+	//+optional
+	KubeletPath string `json:"kubeletPath,omitempty"`
+
+	// KubeletArgs Customized flag for kubelet process
+	// +optional
+	Kubelet ComponentConfig `json:"kubelet,omitempty"`
+
+	// KubeProxyArgs Customized flag for kube-proxy process
+	// +optional
+	KubeProxy ComponentConfig `json:"kubeProxy,omitempty"`
+}
+
+// RKE2ConfigStatus defines the observed state of RKE2Config
+type RKE2ConfigStatus struct {
+	// Ready indicates the BootstrapData field is ready to be consumed
+	Ready bool `json:"ready,omitempty"`
+
+	BootstrapData []byte `json:"bootstrapData,omitempty"`
```

Comment:
can we provider any description? also, why is this stored as a byte array?

Reply:
It looks like this is inherited from K3s provider, I thought it did make sense but this code search suggests that this thing is unnecessary. Do you agree ?

Reply:
Yeah, it looks unnecessary